### PR TITLE
Tezos-opencl: Workaround for buggy nvidia driver

### LINKED ISSUE #5546: tezos-opencl kernel build takes longer than 10 hours with recent nvidia drivers
This was during a `--test=0 --format=opencl` (edit: and a 2080ti) with driver 550.107.02. I killed it after 45 minutes with 100% CPU. Did we see this before?

Also tried `-test=0 -form:tezos-opencl` by itself, same thing but I interrupted sooner. Will try other drivers (as long as they are readily available as Ubuntu packages).

### PR DESCRIPTION
**Pull request by magnumripper**
Closes #5546

### CODE REVIEW

**Review by solardiz — APPROVED** [review]

File: run/opencl/ed25519-donna/ed25519-donna-impl-base.h [hunk 1]
```diff
@@ -121,6 +121,9 @@ ge25519_scalarmult_base_choose_niels(ge25519_niels *t, uint32_t pos, signed char
 
 /* computes [s]basepoint */
 static void
+#if gpu_nvidia(DEVICE_INFO) && DEV_VER_MAJOR > 500
```
Comment (solardiz) [inline run/opencl/ed25519-donna/ed25519-donna-impl-base.h]: The strict `> 500` is a bit weird, meaning 501 or higher. Perhaps you meant `>= 500`.

Comment (magnumripper) [reply run/opencl/ed25519-donna/ed25519-donna-impl-base.h]: I *think* I had it working with 525.xx so maybe we should use that. Also I now confirmed the 560 or higher CUDA drivers work fine without the noinline, which is great! I'll amend it.

File: run/opencl/ed25519-donna/ed25519-donna-impl-base.h [hunk 2]
```diff
@@ -121,6 +121,9 @@ ge25519_scalarmult_base_choose_niels(ge25519_niels *t, uint32_t pos, signed char
 
 /* computes [s]basepoint */
 static void
+#if gpu_nvidia(DEVICE_INFO) && DEV_VER_MAJOR > 525 && DEV_VER_MAJOR < 560
```
Comment (solardiz) [inline run/opencl/ed25519-donna/ed25519-donna-impl-base.h:124]: Again, you probably mean `>= 525`, not `> 525`.

Comment (solardiz) [reply run/opencl/ed25519-donna/ed25519-donna-impl-base.h:124]: Oh, I see you wrote you saw the issue with 535 and 550. But apparently not with 525, nor 560. If so, the check is right.

Comment (ghost) [reply run/opencl/ed25519-donna/ed25519-donna-impl-base.h:124]: Generic, but very appropriate to the situation: it's documenting that anything after 525 and before 560 is or could be bad.

**Pull request merged**